src/pydantic_zarr/experimental/v2.py

```diff
@@ -515,7 +515,7 @@ def from_zarr(cls, group: zarr.Group, *, depth: int = -1) -> Self:
 
         result: GroupSpec
         attributes = group.attrs.asdict()
-        members: dict[str, object] = {}
+        members: dict[str, dict[str, object]] = {}
```

Comment:
Should this not be `dict[str, GroupSpec | ArraySpec]`, and `pydantic` will handle the casting if it's given `dict[str, object]` as input?

Reply:
we are filling `members` with dicts produced by `ArraySpec.model_dump()` / `GroupSpec.model_dump()`. If we don't do that, then pydantic will reject models that declared specialized members, because pydantic uses nominal typing (isinstance checks) when the input is a basemodel